src/gas-price/gas-price.ts

```diff
@@ -81,9 +81,27 @@ export const fetchAndStoreGasPrice = async (
   return gasPrice;
 };
 
-export const getRecommendedGasPrice = (chainId: string, providerName: string, sponsorWalletAddress: Address) => {
+export const getRecommendedGasPrice = (
+  chainId: string,
+  providerName: string,
+  sponsorWalletAddress: Address,
+  dataFeedIds: Hex[]
+) => {
   const state = getState();
-  const pendingTransactionInfo = state.pendingTransactionsInfo[chainId]![providerName]![sponsorWalletAddress];
+  const pendingTransactionInfos = dataFeedIds.reduce((acc: PendingTransactionInfo[], dataFeedId) => {
+    const pendingTransactionInfo =
+      state.pendingTransactionsInfo[chainId]![providerName]![sponsorWalletAddress]?.[dataFeedId];
+    if (!pendingTransactionInfo) {
+      return acc;
+    }
+    return [...acc, pendingTransactionInfo];
+  }, []);
+  // Get the oldest PendingTransactionInfo and if the oldest is not a single object then minBy will return the one with the largest consecutivelyUpdatableCount.
```

Comment:
Probably this comment is reasonably correct but not consistent with the code. Code calculates the oldest pending transaction info according to `firstUpdatableTimestamp` rather than `consecutivelyUpdatableCount`.